ksqldb-functional-tests/src/test/resources/sql-tests/query-upgrades/filters.sql

```diff
@@ -131,16 +125,133 @@ CREATE STREAM j AS SELECT s.id, s.foo, t.bar FROM s JOIN t ON s.id = t.id;
+SET 'ksql.create.or.replace.enabled' = 'true';
+
+CREATE STREAM foo (id INT KEY, col1 INT) WITH (kafka_topic='foo', value_format='JSON');
+CREATE TABLE bar AS SELECT id, COUNT(col1) as count FROM foo WHERE col1 > 0 GROUP BY id;
+
+INSERT INTO foo (id, col1) VALUES (1, 1);
+
+ASSERT VALUES bar (id, count) VALUES (1, 1);
+
+CREATE OR REPLACE TABLE bar AS SELECT id, COUNT(col1) as count FROM foo GROUP BY id;
+
+INSERT INTO foo (id, col1) VALUES (1, 0);
+
+ASSERT VALUES bar (id, count) VALUES (1, 2);
+
+----------------------------------------------------------------------------------------------------
+--@test: change filter in StreamAggregate to another column that already exists in input
+----------------------------------------------------------------------------------------------------
+SET 'ksql.create.or.replace.enabled' = 'true';
+
+CREATE STREAM foo (id INT KEY, col1 INT) WITH (kafka_topic='foo', value_format='JSON');
+CREATE TABLE bar AS SELECT id, COUNT(col1) as count FROM foo WHERE col1 >= 0 GROUP BY id;
+
+INSERT INTO foo (id, col1) VALUES (1, 0);
+INSERT INTO foo (id, col1) VALUES (2, 0);
+
+ASSERT VALUES bar (id, count) VALUES (1, 1);
+ASSERT VALUES bar (id, count) VALUES (2, 1);
+
+CREATE OR REPLACE TABLE bar AS SELECT id, COUNT(col1) as count FROM foo WHERE id > 1 GROUP BY id;
+
+INSERT INTO foo (id, col1) VALUES (1, 0);
+INSERT INTO foo (id, col1) VALUES (2, -1);
+
+ASSERT VALUES bar (id, count) VALUES (2, 2);
+
+----------------------------------------------------------------------------------------------------
+--@test: remove filter in StreamAggregate where columns are not already in input schema
+--@test: add filter in StreamAggregate where columns are not in input schema
 --@expected.error: io.confluent.ksql.util.KsqlException
---@expected.message: Upgrades not yet supported for StreamAggregate
+--@expected.message: StreamAggregate must have matching nonAggregateColumns. Values differ: [`ID`, `ROWTIME`, `COL1`] vs. [`ID`, `ROWTIME`]
 ----------------------------------------------------------------------------------------------------
 SET 'ksql.create.or.replace.enabled' = 'true';
 
 CREATE STREAM foo (id INT KEY, col1 INT) WITH (kafka_topic='foo', value_format='JSON');
+CREATE TABLE bar AS SELECT id, COUNT(*) as count FROM foo WHERE col1 > 0 GROUP BY id;
+
+CREATE OR REPLACE TABLE bar AS SELECT id, COUNT(*) as count FROM foo GROUP BY id;
 
+----------------------------------------------------------------------------------------------------
+-- until we think this through a little bit more, don't allow changing non-aggregate columns
+-- to StreamAggregate nodes, though this should technically be OK
+
+--@test: add filter in StreamAggregate where columns are not in input schema
+--@expected.error: io.confluent.ksql.util.KsqlException
+--@expected.message: StreamAggregate must have matching nonAggregateColumns. Values differ: [`ID`, `ROWTIME`] vs. [`ID`, `COL1`]
```

Comment:
this is kind of a confusing error message since the columns are part of the aggregate, and the user likely doesn't know about the implicit substitution of wildcard for ROWTIME. I think its fair to say fixing this is probably out of scope for this PR, but maybe file an issue?

Reply:
yeah, I need to improve the error messages across the board for query upgrades when physical plan checks fail. I'm not entirely sure what a good strategy for that is but I'll file a ticket

Reply:
https://github.com/confluentinc/ksql/issues/6087

Reply:
and specifically for fixing the rowtime issue: https://github.com/confluentinc/ksql/issues/3732